Library/Homebrew/compilers.rb

```diff
@@ -2,8 +2,8 @@
 # frozen_string_literal: true
 
 module CompilerConstants
-  GNU_GCC_VERSIONS = %w[4.9 5 6 7 8 9 10 11 12 13 14].freeze
-  GNU_GCC_REGEXP = /^gcc-(4\.9|[5-9]|10|11|12|13|14)$/
+  GNU_GCC_VERSIONS = %w[7 8 9 10 11 12 13 14].freeze
```

Comment:
4.9 (removed) and 5 & 6 (planned to remove in October). May need to check on which LTS distros this will impact.

Reply:
Rough estimates for some common ones. A number of others like Rocky/Centos/etc are RHEL-based (and also some of the most problematic ones based on issues):

| Distro | EOL? | GCC |
| --- | --- | --- |
| Debian 8 | 2025 | GCC 8 |
| Ubuntu 16.04 | 2026 | GCC 5 |
| Ubuntu 18.04 | 2028 | GCC 7 |
| RHEL 7 | 2026 | GCC 4.8 |
| RHEL 8 | 2032 | GCC 8 |

Reply:
> Do we have analytics for GCC actually used?

Unfortunately not. Stripping out arg values is intended behaviour because they are not validated so could contain any input. Though `--cc` isn't the only way it can be used as formulae can do e.g. `ENV.public_send(:"gcc-5")` (horrible API we should really fix but whatever).